packages/core/src/payment/strategies/worldpayaccess/worldpayaccess-payment-strategy.ts

```diff
@@ -186,4 +211,39 @@ export default class WorldpayaccessPaymetStrategy extends CreditCardPaymentStrat
 
         return iframe;
     }
+
+    private _submitAdditionalAction() {
+        if (this._shouldRenderHostedForm) {
+            if (!this._hostedForm || !this._hostedForm.submit) {
+                throw new NotInitializedError(
+                    NotInitializedErrorType.PaymentNotInitialized
+                );
+            }
+
+            const hostedForm = this._hostedForm;
+
+            return async (paymentPayload: OrderPaymentRequestBody) => {
+                await hostedForm.submit(paymentPayload);
+                return this._store.getState();
+            }
+        } else {
+            return async (paymentPayload: OrderPaymentRequestBody) => {
+                return await this._store.dispatch(this._paymentActionCreator.submitPayment( paymentPayload ));
+            }
```

Comment:
This feels a bit weird, can we try an use promise here?

Reply:
Why is weird? I feel that a promise is going to overcomplicate the code and only we need the right function to be execute in line 67
![image](https://user-images.githubusercontent.com/4907027/190227715-0b867e88-6383-44f2-917b-915aab187e56.png)